main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateFareRuleZoneIdFieldsValidator.java

```diff
@@ -77,9 +78,18 @@ public void validate(NoticeContainer noticeContainer) {
    * @return the hashcode associated to the combination of this {@code GtfsFareRule} "route_id",
    *     "origin_id", "contains_id" and "destination_id".
    */
-  private int getHash(GtfsFareRule fareRule) {
-    return Objects.hash(
-        fareRule.routeId(), fareRule.originId(), fareRule.containsId(), fareRule.destinationId());
+  private static long getHash(GtfsFareRule fareRule) {
+    return HASH_FUNCTION
+        .newHasher()
+        .putUnencodedChars(fareRule.routeId())
+        .putChar('\0')
```

Comment:
Why using this here?

Reply:
You see, putUnencodedChars (and putString as well) do not put any delimiter, so

```
newHasher().putUnencodedChars("ab").putUnencodedChars("c").hash()
newHasher().putUnencodedChars("a").putUnencodedChars("bc").hash()
```

produce the same result.

Valid GTFS strings never include '\0', so I am using it as a delimiter.

Reply:
Oh! Makes sense! Thanks.